### PR TITLE
Dashboard: Fix/admin console warnings

### PR DESCRIPTION
**Pull request by coder-karen**
#### Changes proposed in this Pull Request:

* This PR fixes the following React-related console error message that displays on the Jetpack dashboard - `wp-admin/admin.php?page=jetpack#/dashboard`- with script debug enabled:

```
A future version of React will block javascript: URLs as a security precaution. Use event handlers instead if you can. If you need to generate unsafe HTML try using dangerouslySetInnerHTML instead. React was passed "javascript:void(0)".
```

* It does this by changing the link that is passed to be '#!', and in the process also change the a link to be a Button component (changed from a Link component in an earlier revision). The reasoning is explained in the (now striked out) text below, but in short since the link isn't an actual link a button worked better. This did involve a little styling as there didn't appear to be options available by default to make the button look like a regular link.
~* The Link component was used primarily because regular anchor links aren't really ideal for a link that doesn't actually go anywhere (technically a button would be better as I understand it), and even though the Link component renders as an anchor element it does provide extra features and benefits if needed later.~
~* Other options not included here:~
  ~* Another option was to change the links to button elements, which involved adding styling to style.scss. That would mean that should link styles change throughout the dashboard, the button 'links' would need to manually be re-styled to be the same as other links.~
  ~* Alternatively, using `event.preventListener` was a third option to prevent the link from going anywhere, before running our onClick function.~

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

n/a

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* On a Jetpack-connected site, enable script debugging in `wp-config.php`: `define( 'SCRIPT_DEBUG', true );`.
* With the browser developer console open, visit the Jetpack dashboard at `wp-admin/admin.php?page=jetpack#/dashboard`.
* You should see the console error mentioned above.
* Apply this PR and repeat - this time that specific console error should not be showing anymore.

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/at-a-glance/akismet.jsx [hunk 1]
```diff
@@ -99,9 +100,9 @@ class DashAkismet extends Component {
 
 		const getAkismetUpgradeBanner = () => {
 			const description = createInterpolateElement(
-				__( 'Already have a key? <a>Activate Akismet</a>', 'jetpack' ),
+				__( 'Already have a key? <Link>Activate Akismet</Link>', 'jetpack' ),
 				{
-					a: <a href="javascript:void(0)" onClick={ this.onActivateClick } />,
+					Link: <Link to="#!" onClick={ this.activateVideoPress } />,
```
Comment (samiff) [inline projects/plugins/jetpack/_inc/client/at-a-glance/akismet.jsx]:
```suggestion
					Link: <Link to="#!" onClick={ this.onActivateClick } />,
```

**Comment by samiff** [comment]
> regular anchor links aren't really ideal for a link that doesn't actually go anywhere (technically a button would be better as I understand it)

That is what I understand as well. Would hate for this PR to turn into an accessibility audit of all dashboard links, but I wonder if we already have styling for buttons (to make them look like links) elsewhere to leverage?

File: projects/plugins/jetpack/_inc/client/at-a-glance/akismet.jsx (hunk 1 above)
Comment (coder-karen) [reply projects/plugins/jetpack/_inc/client/at-a-glance/akismet.jsx]: Thanks for catching that.

**Comment by coder-karen** [comment]
> Would hate for this PR to turn into an accessibility audit of all dashboard links, but I wonder if we already have styling for buttons (to make them look like links) elsewhere to leverage?

Oh yes, thank you! It looks like I may be able to use our own internal Button component for this.

**Review by samiff — COMMENTED** [review]
Updated ~links~ buttons look good 👍 Seems like there is one more that could be updated here:

https://github.com/Automattic/jetpack/blob/8a4fa527271ff18a76f497916c6827e96c7e9d0f/projects/plugins/jetpack/_inc/client/at-a-glance/stats/index.jsx#L246

And the stats one at `/wp-admin/admin.php?page=jetpack#/traffic` isn't picking up the CSS styling since it lacks the `.jp-dash-item__description` class that `.jp-link-button` is nested under.

**Comment by coder-karen** [comment]
Thanks for catching that. I've added the additional button and updated CSS classes. One thing I haven't been able to test is VaultPress (`scan.jsx`) notification , so I'm not sure if that is for legacy users. Presumably it would be similar to the Akismet one if so, and should be picked up by the same CSS selectors.

**Review by samiff — APPROVED** [review]

**Pull request merged**